docs/developer-env.md

```diff
@@ -0,0 +1,534 @@
+
+Nextflow must be installed separately. See {ref}`install-page` for Nextflow installation instructions.
+
+:::{note}
+If you are using a Windows computer, first install and configure the Windows Subsystem for Linux (WSL). See {ref}`wsl` for installation instructions.
+:::
+
+(vscode-install)=
+
+## VS Code
+
+An Integrated Development Environment (IDE) provides a user-friendly interface for writing, editing, and managing code. Installing one is an essential step for setting up your environment. 
+
+Visual Studio Code (VS Code) is a popular lightweight IDE known for its versatility and extensibility. It offers features like syntax highlighting, intelligent code completion, and integrated debugging tools for various programming languages. VS Code supports Windows, macOS, and Linux, and is a good choice for both new and experienced Nextflow developers.
+
+````{tabs}
+
+```{group-tab} Windows
+
+To install VS Code on Windows:
+
+1. Visit the [VS Code](https://code.visualstudio.com/download) website.
+1. Download VS Code for Windows.
+1. Double-click the installer executable (`.exe`) file and follow the set up steps.
+
+```
+
+```{group-tab} macOS
+
+To install VS Code on macOS:
+
+1. Visit the [VS Code](https://code.visualstudio.com/download) website.
+1. Download VS Code for macOS.
+1. Drag the `Visual Studio Code.app` application to the Applications folder to add it to the macOS Launchpad.
+
+```
+
+```{group-tab} Linux
+
+To install VS Code on Linux Debian/Ubuntu distributions:
+
+1. Visit the [VS Code](https://code.visualstudio.com/download) website.
+1. Download the VS Code Linux Debian/Ubuntu (`.deb`) distribution.
+1. Open a new terminal window.
+1. Navigate to the folder where you downloaded VS Code.
+1. Run `sudo apt install ./<file>.deb`, replacing `<file>` with the full file name.
+   
+   :::{note}
+   If you're using an older Linux distribution, run `sudo dpkg -i <file>.deb` to install VS Code and `sudo apt-get install -f` to install dependencies.
+   :::
+   
+See [Linux installation](https://code.visualstudio.com/docs/setup/linux#_installation) for information about installing VS Code on other distributions.
+
+```
+
+````
+
+## VS Code Extensions
+
+Extensions are a key feature of IDEs and allow you to customize your development environment by adding support for various programming languages, tools, and features. The [VS Code Marketplace](https://marketplace.visualstudio.com/vscode) offers thousands of extensions that can enhance your productivity and tailor the editor to your specific needs.
+
+
+### Nextflow
+
+The VS Code [Nextflow extension](https://marketplace.visualstudio.com/items?itemName=nextflow.nextflow) adds Nextflow language support to the editor. The Nextflow extension enhances development with:
+
+- Diagnostics
+- Hover hints
+- Code navigation
+- Code completion
+- Formatting
+- Renaming
+- Parameter schemas
+- DAG previews
+
+See {ref}`vs-code-page` for more information about the Nextflow extension.
+
+````{tabs}
```

Comment:
Can we have just one of these for "installing VSCode extensions"? Feels bulky to have essentially the identical content repeated for each set of VSCode extensions.

At the moment the VSCode section is like, most of the page. I think making each subsection shorter would help.

Reply:
Done.